### PR TITLE
Bug fix panic tps benchmark

### PR DESCRIPTION
**Pull request by miiu96**
Fix concurrent access on map that store shard statistics information for TPS Benchmark. 
When method ShardStatistics() is called will return a copy of map that contains shard statistics.

### CODE REVIEW

File: core/statistics/tpsBenchmark.go [hunk 1]
```diff
@@ -179,8 +179,13 @@ func (s *TpsBenchmark) NrOfShards() uint32 {
 // ShardStatistics returns the current statistical state for a given shard
 func (s *TpsBenchmark) ShardStatistics() map[uint32]ShardStatistic {
 	s.mut.RLock()
-	defer s.mut.RUnlock()
-	return s.shardStatistics
+	copyMapShardStatistics := make(map[uint32]ShardStatistic)
```
Comment (andreibancioiu) [inline core/statistics/tpsBenchmark.go:182]: Seems fine, I see that `updateStatistics` is protected by the same lock.

**Review by andreibancioiu — APPROVED** [review]

File: core/statistics/tpsBenchmark.go (hunk 1 above)
Comment (bogdan-rosianu) [reply core/statistics/tpsBenchmark.go:182]: rename to `shardStatisticsMapCopy`

File: core/statistics/tpsBenchmark_test.go [hunk 2]
```diff
@@ -636,3 +636,47 @@ func TestTpsBenchmark_ZeroTxMetaBlockAndEmptyShardHeader(t *testing.T) {
 	bigTxCount := big.NewInt(0)
 	assert.Equal(t, bigTxCount, tpsBenchmark.TotalProcessedTxCount())
 }
+
+func TestTpsBenchmark_ShardStatisticConcurrentAccess(t *testing.T) {
+	t.Parallel()
+
+	tpsBenchmark, _ := statistics.NewTPSBenchmark(12, 4)
```
Comment (bogdan-rosianu) [inline core/statistics/tpsBenchmark_test.go:643]: use recover() as you should check that no panic has encountered during the test

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: core/statistics/tpsBenchmark_test.go (hunk 2 above)
Comment (miiu96) [reply core/statistics/tpsBenchmark_test.go:649]: done.

File: core/statistics/tpsBenchmark.go (hunk 1 above)
Comment (miiu96) [reply core/statistics/tpsBenchmark.go]: done.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed

**Pull request merged**